broker-plugins/jdbc-store/src/main/java/org/apache/qpid/server/store/jdbc/AbstractJDBCMessageStore.java

```diff
@@ -230,6 +233,8 @@ public Thread newThread(final Runnable r)
         });
         _executor.prestartAllCoreThreads();
 
+        _contextKeys = parent.getContextKeys(false);
+        _inClauseMaxSize = getContextValue(Integer.class, IN_CLAUSE_MAX_SIZE, IN_CLAUSE_MAX_SIZE_DEFAULT);
```

Comment:
You're implicitly setting an order between these 2 lines since getContextValue uses _contextKeys. Maybe _contextKeys should be a parameter of getContextValue?

Reply:
Actually, I think that adding field '_contextKeys' is premature optimization. I will delete it